### PR TITLE
chore: add extract all authors methods

### PR DESCRIPTION
**Pull request by Kawacrepe**
Add method "extractAllAuthorsFromLibrary" to extract all authors for a given scanner payload

### CODE REVIEW

**Comment by tony-go** [comment]
Yo @Kawacrepe 👋 

WDYT about adding a few tests?

**Comment by Kawacrepe** [comment]
Hey @tony-go it's on my todo list, I just didn't know if my PR was correct and if it would have been better if someone else give an eye on the repository, so I wanted feedback :)

File: index.d.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
 // Import Third-party Dependencies
 import { Maintainer } from "@npm/types";
+import { Scanner } from "@nodesecure/scanner";
 
-export function extractAndOptimizeUsers(maintainers: Maintainer): Maintainer[] | [];
\ No newline at end of file
+export function extractAndOptimizeUsers(maintainers: Maintainer): Maintainer[] | [];
+export function extractAllAuthorsFromLibrary(library: Scanner.Payload): Maintainer[] | [];
```
Comment (fraxken) [inline index.d.ts:6]: Need a new line at the end

File: src/index.js [hunk 2]
```diff
@@ -15,12 +12,35 @@ export async function extractAndOptimizeUsers(dependencyMetadata) {
...
+
+  const authors = [];
+  for (const dep of Object.values(library.dependencies)) {
+    const { author, maintainers, publishers } = dep.metadata;
+
+    const authorsFound = formatResponse(author, maintainers, publishers);
+    if (authorsFound) {
+      for (const author of authorsFound) {
+        authors.push(author);
+      }
```
Comment (fraxken) [inline src/index.js:28]: `authors.push(...authorsFound);`

No need to execute a for iteration.

Comment (fraxken) [reply src/index.js:28]: Also if authorsFound is returned "empty" no need to protect with `if` condition.

**Review by fraxken — CHANGES_REQUESTED** [review]

**Comment by fraxken** [comment]
Could be useful to add UT not just for the main entry point but also for functions in leveinstein etc.

**Comment by Kawacrepe** [comment]
I added some UT on internal functions which helped me to find a bug in my implementation (if there were the same author in maintainer + publisher I didn't saved `version` & `at` properties).

**Pull request merged**